Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2192 @@
+</p>
+
+
+<p>
+Because Fortran 2003 does specify support for a special <code>FINAL</code> procedure to clean up local or dynamic variables, the call to <code>release()</code> can be replaced by adding a <code>FINAL</code> procedure. The SWIG Fortran interface can generate this procedure, which will call the C++ destructor:
+</p>
+
+
+<div class="code"><pre><code>%feature(&quot;final&quot;) Foo;
+%include &quot;Foo.h&quot;
+</code></pre>
+</div>
+
+
+<p>
+<strong>However</strong>, this feature is relatively untested and its behavior could be compiler-dependent, so extreme caution is recommended when enabling it.
+</p>
+
+<H3><a name="Fortran_member_functions">Member functions</a></H3>
+
+<p>
+SWIG generates unique, private procedure names (generally <code>swigf_{classname}_{funcname}</code> for each class and function. These procedures are bound to the type. If <a href="#Fortran_function_overloading">function overloading</a> is used, “generic” procedures will be added to the derived type.
+</p>
+
+
+<p>
+Type-bound procedures in Fortran proxy classes are treated exactly the same as for native derived types:
+</p>
+
+
+<div class="targetlang"><pre><code>integer(C_INT) :: value
+type(Foo) :: food
+food = Foo()
+call food%do_something()
+value = food%get_something()
+</code></pre>
+</div>
+
+
+<p>
+Function overloading for derived types is implemented using <em>generic interfaces</em>. Each overloaded function gets a unique internal symname, and they are bound together in a generic interface. For example, if a member function <code>doit</code> of class <code>Action</code> is overloaded, a generic binding will be generated inside the Fortran proxy derived type:
+</p>
+
+
+<div class="targetlang"><pre><code>  procedure, private :: doit__SWIG_0 =&gt; swigf_Action_doit__SWIG_0
+  procedure, private :: doit__SWIG_1 =&gt; swigf_Action_doit__SWIG_1
+  generic :: doit =&gt; doit__SWIG_0, doit__SWIG_1
+</code></pre>
+</div>
+
+
+<p>
+As with <a href="#Fortran_function_overloading">free functions</a>, a member function returning <code>void</code> cannot be overloaded with a function returning non-void.
+</p>
+
+<H3><a name="Fortran_member_data">Member data</a></H3>
+
+<p>
+SWIG generates member functions for class member data in the same way that it generates free functions for <a href="#Fortran_global_variables">global variables</a>. Each public member produces a “getter”, and unless the data is marked <code>const</code>, it generates a “setter”.
+</p>
+
+
+<p>
+For a struct
+</p>
+
+<pre class="c++"><code>struct Foo {
+  int val;
+};
+</code></pre>
+
+<p>
+the interface to an instance and its data is:
+</p>
+
+
+<div class="targetlang"><pre><code>type(Foo) :: f
+f = Foo()
+call 
```

Comment:
It serves the same purpose, but since it's not a boolean value, I thought `mem` was better than `memown = [OWN|MOVE|REF|CREF|NULL]`. I was originally going to name it `owner` like the special variable `$owner`, but thought that `swigdata%owner = REF` was confusing. Yep, `cptr` is definitely a better choice, I'll change that.

Reply:
I can only find one instance of mem being mentioned in the docs (it's not an easy name to search for in an html file). I still think memown is still a better name given your description of it:
```
 self%swigdata%mem that describes memory ownership
```
I can't see any mention in the docs of the acceptable values being `[OWN|MOVE|REF|CREF|NULL]` that you mentioned above.

Reply:
These are documented in the "Memory management" section under "Proxy classes". If the name needs to be changed to `memown` for greater consistency with C#/Java, let's defer that till after the merge.

Reply:
I've changed the names of these to `swigdata%cptr` and `swigdata%cmemflags`.